Jenkinsfile

```diff
@@ -125,26 +125,52 @@ pipeline {
                         label env.VDVS_65_NODE_ID
                     }
 
-                steps {
-                    sh "echo Test VDVS On 6.5 setup"
-                    sh "echo ESX=$ESX; echo VM1=$VM1; echo VM2=$VM2; echo VM3=$VM3; echo PKG_VERSION"
-                    sh "echo starting e2e tests" 
-                    sh "make test-e2e"
-                    sh "make test-esx"
-                    sh "make test-vm"
+                    steps {
+		        script {
+			    try {
+                                sh "echo Test VDVS On 6.5 setup"
+                                sh "echo ESX=$ESX; echo VM1=$VM1; echo VM2=$VM2; echo VM3=$VM3; echo PKG_VERSION"
+                                sh "echo starting e2e tests"
+                                sh "make test-e2e"
+                                sh "make test-esx"
+                                sh "make test-vm"
+			    }
+			    catch (ex) {
+                                sh "ssh ${env.GOVC_USERNAME}@$VM1 ${stopContainer}; ${removeContainer}; ${removeVolume}"
+                                sh "ssh ${env.GOVC_USERNAME}@$VM2 ${stopContainer}; ${removeContainer}; ${removeVolume}"
+                                sh "ssh ${env.GOVC_USERNAME}@$VM3 ${stopContainer}; ${removeContainer}; ${removeVolume}"
+                                sh "make clean-all"
```

Comment:
It would be good to have the `sh` cleanup statements in a single function which can be called from necessary locations. We could possibly also pass in a boolean parameter to enable/disable `make clean-vfile` as needed.